blob/s3blob/s3blob.go

```diff
@@ -129,53 +129,14 @@ func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket
 	configProvider := &gcaws.ConfigOverrider{
 		Base: o.ConfigProvider,
 	}
-	overrideCfg, err := o.forParams(ctx, u.Query())
+	overrideCfg, err := gcaws.ConfigFromURLParams(u.Query())
 	if err != nil {
 		return nil, fmt.Errorf("open bucket %v: %v", u, err)
 	}
-	if overrideCfg != nil {
-		configProvider.Configs = append(configProvider.Configs, overrideCfg)
-	}
+	configProvider.Configs = append(configProvider.Configs, overrideCfg)
 	return OpenBucket(ctx, configProvider, u.Host, &o.Options)
 }
 
-var legalQueryParam = map[string]bool{
```

Comment:
Seems useful to disallow unrecognized query parameters. Consider adding a function in `aws` that would indicate whether a query parameter would be recognized by the common set.

Reply:
There's a `default` clause that returns an error for unrecognized query parameters; this avoids having a separate `map` of them.

The expectation is that if the caller has additional query parameters that they want to process, they'll do that first and remove them from `url.Values` before calling this function. (`s3blob` doesn't have any additional query parameters though).

Reply:
(added a comment to this effect)

Reply:
Thanks, I missed that on first read.